### PR TITLE
Add encryption meta to metapd.Region

### PR DESCRIPTION
**Pull request by yiwu-arbug**
Adding encryption meta to `metapb.Region`. This is going to be used by PD to encrypt the start_key and end_key when persisting the `Region` struct.

Signed-off-by: Yi Wu <yiwu@pingcap.com>

### CODE REVIEW

File: proto/encryptionpb.proto [hunk 1]
```diff
@@ -11,6 +11,16 @@ option (rustproto.lite_runtime_all) = true;
 
 option java_package = "org.tikv.kvproto";
 
+// General encryption metadata for any data type.
+message EncryptionMeta {
+    // Whether data is encrypted.
+    bool is_encrypted = 1;
```
Comment (HunDunDM) [inline proto/encryptionpb.proto:17]: `EncryptionMeta` is not `nil`, but `is_encrypted` is `false`. What is the corresponding situation?

**Review by HunDunDM — COMMENTED** [review]
rest LGTM

File: proto/encryptionpb.proto (hunk 1 above)
Comment (yiwu-arbug) [reply proto/encryptionpb.proto:17]: The two case should both treated as the region boundary keys are not encrypted.

**Comment by yiwu-arbug** [comment]
updated. PTAL

**Review by HunDunDM — APPROVED** [review]
LGTM

**Review by Yisaer — APPROVED** [review]
LGTM

**Pull request merged**